src/targets/shell/curl.js

```diff
@@ -58,10 +58,17 @@ module.exports = function (source, options) {
     case 'application/x-www-form-urlencoded':
       if (source.postData.params) {
         source.postData.params.map(function (param) {
-          code.push(
-            '%s %s', opts.binary ? '--data-binary' : (opts.short ? '-d' : '--data'),
-            helpers.quote(util.format('%s=%s', param.name, param.value))
-          )
+          if (opts.binary) {
+            code.push(
+              '%s %s', '--data-binary',
+              helpers.quote(util.format('%s=%s', param.name, param.value))
+            )
+          } else {
+            code.push(
+              '%s %s', '--data-urlencode',
+              helpers.quote(util.format('=%s=%s', param.name, param.value))
```

Comment:
I don't think this preceding `=` should be here.

```suggestion
              helpers.quote(util.format('%s=%s', param.name, param.value))
```

Reply:
It was intended. To tell to cUrl to URL encode the data, we should use `--data-urlencode`. This will encode the value, but we should also encode the key. It can be done by prefixing the string to encode with `=`.

Extract from this [page](https://ec.haxx.se/http/http-post#url-encoding):

>If you want to URL encode the field name as well, like if you want to pass on a field name called "user name", you can ask curl to encode the entire string by prefixing it with an equals sign (that will not get sent):
```
curl --data-urlencode "=user name=John Doe (Junior)" http://example.com
```

Reply:
It was intended to let cUrl URL encodes the data. I was reading this [page](https://ec.haxx.se/http/http-post#url-encoding):

> If you want to URL encode the field name as well, like if you want to pass on a field name called "user name", you can ask curl to encode the entire string by prefixing it with an equals sign (that will not get sent):

Reading it one more time has started to make me doubt 🤔 

The following

```
curl --request POST \
  --url http://localhost:3000/url-encoded \
  --header 'Content-Type: application/x-www-form-urlencoded' \
  --data-urlencode '=$filter=my&value = 1'
```

produces the encoded string: `%24filter%3Dmy%26value%20%3D%201` 
and this is interpreted in `{"$filter=my&value = 1":""}`

Removing the `=` will let cUrl to encode the value correctly. However, we should also encode the `key`. I've thought cUrl would be able to do it for us, but it seems it won't. Therefore I will encode the key "manually."

Reply:
I also am not sure about this.  I think given that we don't know, the best case is to leave it "as it was" and see if that works for people.

I'll leave this unresolved so that if you, the reader (some time in the future), are experiencing this: you'll see what our reasoning was.  If this presents you a problem please open a GitHub issue!